app/services/active_sale/line_item_updater.rb

```diff
@@ -0,0 +1,23 @@
+module ActiveSale
+  class LineItemUpdater < BaseService
+    def call
+      Spree::LineItem.transaction do
```

Comment:
@sokmesakhiev we would need to update the cart too so we also need to find pending carts that have line_items product changes.

discount_changes:
- product_2

cart1:
 - product_1 
 - product_2 

cart2: 
- product_3
- product_1

cart3: 
- product_3
- product_2
- product_1

so we need to update cart1 and cart3 because product_2 affected the price: